src/game/game.c

```diff
@@ -245,51 +247,51 @@ const u16 gUnknown_080D5964[][2] = {
     { 32, 232 }, { 32, 264 }, { 32, 264 }, { 32, 264 },
 };
 
-void sub_801C774(void);
+void StageInit_Zone1(void);
 void sub_801CB74(void);
 void SetupSpotlightSnowAndCreateSpotlights(void);
-void sub_801CEE4(void);
-void sub_801D104(void);
-void sub_801D1A8(void);
-void sub_801D95C(void);
-void sub_801DF08(void);
-void sub_801E118(void);
-void sub_801E12C(void);
+void StageInit_Zone5(void);
+void StageInit_Zone6_Acts(void);
+void StageInit_Zone6_Boss(void);
+void StageInit_Zone7(void);
+void StageInit_ZoneFinal_0(void);
+void StageInit_Zone2(void);
+void nullsub_801E12C(void);
 const VoidFn gUnknown_080D5988[] = {
-    sub_801C774,
-    sub_801C774,
-    sub_801C774,
-    NULL, // Anti-Formatting
-    sub_801E118,
-    sub_801E118,
-    sub_801E118,
-    NULL, //
-    sub_801CB74,
-    sub_801CB74,
-    sub_801CB74,
-    NULL, //
-    SetupSpotlightSnowAndCreateSpotlights,
-    SetupSpotlightSnowAndCreateSpotlights,
-    sub_801E12C,
-    NULL, //
-    sub_801CEE4,
-    sub_801CEE4,
-    NULL,
-    NULL, //
-    sub_801D104,
-    sub_801D104,
-    sub_801D1A8,
-    NULL, //
-    sub_801D95C,
-    sub_801D95C,
-    sub_801D95C,
-    NULL, //
-    NULL,
-    sub_801DF08,
-    sub_801C774,
-    NULL, //
-    NULL,
-    NULL,
+    [LEVEL_INDEX(ZONE_1, ACT_1)] = StageInit_Zone1,
+    [LEVEL_INDEX(ZONE_1, ACT_2)] = StageInit_Zone1,
+    [LEVEL_INDEX(ZONE_1, ACT_BOSS)] = StageInit_Zone1,
+    [LEVEL_INDEX(ZONE_1, ACT_UNUSED)] = NULL, // Anti-Formatting
+    [LEVEL_INDEX(ZONE_2, ACT_1)] = StageInit_Zone2,
+    [LEVEL_INDEX(ZONE_2, ACT_2)] = StageInit_Zone2,
+    [LEVEL_INDEX(ZONE_2, ACT_BOSS)] = StageInit_Zone2,
+    [LEVEL_INDEX(ZONE_2, ACT_UNUSED)] = NULL, //
+    [LEVEL_INDEX(ZONE_3, ACT_1)] = sub_801CB74,
+    [LEVEL_INDEX(ZONE_3, ACT_2)] = sub_801CB74,
+    [LEVEL_INDEX(ZONE_3, ACT_BOSS)] = sub_801CB74,
+    [LEVEL_INDEX(ZONE_3, ACT_UNUSED)] = NULL, //
+    [LEVEL_INDEX(ZONE_4, ACT_1)] = SetupSpotlightSnowAndCreateSpotlights,
+    [LEVEL_INDEX(ZONE_4, ACT_2)] = SetupSpotlightSnowAndCreateSpotlights,
+    [LEVEL_INDEX(ZONE_4, ACT_BOSS)] = nullsub_801E12C,
+    [LEVEL_INDEX(ZONE_4, ACT_UNUSED)] = NULL, //
+    [LEVEL_INDEX(ZONE_5, ACT_1)] = StageInit_Zone5,
+    [LEVEL_INDEX(ZONE_5, ACT_2)] = StageInit_Zone5,
+    [LEVEL_INDEX(ZONE_5, ACT_BOSS)] = NULL,
+    [LEVEL_INDEX(ZONE_5, ACT_UNUSED)] = NULL, //
+    [LEVEL_INDEX(ZONE_6, ACT_1)] = StageInit_Zone6_Acts,
+    [LEVEL_INDEX(ZONE_6, ACT_2)] = StageInit_Zone6_Acts,
+    [LEVEL_INDEX(ZONE_6, ACT_BOSS)] = StageInit_Zone6_Boss,
+    [LEVEL_INDEX(ZONE_6, ACT_UNUSED)] = NULL, //
+    [LEVEL_INDEX(ZONE_7, ACT_1)] = StageInit_Zone7,
+    [LEVEL_INDEX(ZONE_7, ACT_2)] = StageInit_Zone7,
+    [LEVEL_INDEX(ZONE_7, ACT_BOSS)] = StageInit_Zone7,
+    [LEVEL_INDEX(ZONE_7, ACT_UNUSED)] = NULL, //
+    [LEVEL_INDEX(ZONE_FINAL, ACT_XX_FINAL_ZONE)] = NULL, // This doesn't make sense...
+    [LEVEL_INDEX(ZONE_FINAL, ACT_TRUE_AREA_53)] = StageInit_ZoneFinal_0,
```

Comment:
You'll need to verify if `ACT_TRUE_AREA_53` and `ACT_XX_FINAL_ZONE` are the right values?

Reply:
I mean, they've been correct everywhere else, so they should be here, too? :o

Reply:
Very weird then?

Reply:
Yeah